libfwupdplugin/fu-bytes.c

```diff
@@ -181,6 +181,34 @@ fu_bytes_compare(GBytes *bytes1, GBytes *bytes2, GError **error)
 	return fu_memcmp_safe(buf1, bufsz1, 0x0, buf2, bufsz2, 0x0, MAX(bufsz1, bufsz2), error);
 }
 
+/**
+ * fu_bytes_unpad:
+ * @bytes: data blob
+ * @padval: the padding byte to remove, typically 0x00 or 0xFF
+ *
+ * Unpads a GBytes of trailing data.
+ *
+ * Returns: (transfer full): a data blob
+ *
+ * Since: 1.9.21
```

Comment:
we don't have the stream stuff in .21 I thought

Reply:
Ohh indeed. I was going to backport the symbol, but maybe it doesn't make sense. I'd be fine for 2.0.0

Reply:
Maybe I'll rework it to do the right [streaming] thing. Otherwise it's a hack on a hack.

Reply:
I've reworked it and fixed a few bugs on the journey too.